src/main/java/net/pms/dlna/DLNAMediaInfo.java

```diff
@@ -217,6 +217,12 @@ public boolean isExternalSubsParsed() {
 	private byte[] h264_annexB;
 
 	/**
+	 * @deprecated Use standard getter and setter to access this variable.
+	 */
+	@Deprecated
+	public String truncated;
+
+	/**
 	 * Not stored in database.
```

Comment:
@Sami32 We don't make new deprecated fields, instead we make them ```private``` or ```protected``` depending on if the class should be possible to subclass or not. If you don't know, look at the others - you see here that a lot are ```private``` which means you can use ```private``` as well. 

The point with marking something as ```@Deprecated``` is to signal to other classes that use this method or field that it is deprecated and will disappear in a future version. They should update their code to use the alternative (often indicated in the JavaDoc) field or method to achieve the same. In UMS' case it would be no need to use deprecated if it were not for the plugins, it's easy to check that no part of our own code is using a given field or method. Eclipse and other IDE's will happly show you this, and the compilation will fail if you make it ```private``` and it is accessed from outside the class itself. 

To understand ```private```, ```protected``` and ```public``` you'd really need to study the basic concepts of [object oriented programming](https://fr.wikipedia.org/wiki/Programmation_orient%C3%A9e_objet), but in short you can say that:
* ```private``` methods and fields can only be accessed from within the class itself.
* ```protected``` methods and fields can be accessed from within the class itself and from subclasses.
* ```public``` methods and fields can be accessed by any class.

In addition, Java has something I find really stupid called "package private" where only classes in the same "package" ( "Java path") have access. To make it even worse, there is no "word" for this, instead this is the mode it is if you don't write one of the others. A lof of trouble is caused by this "fantastic" invention because it makes it easy for the original developer to work with (since all the related classes often are in the same "package"), but impossible to extend for others.

The "standard solution" is to have the field/class variable ```private``` or ```protected``` to prevent outside classes from referencing (or manipulating) it directly, and then supply ```public``` getXXX and setXXX methods. This has a lot of advantages while almost the only disadvantage is that requires more code. In the real world this doesn't really matter though, as IDE's will create the getters and setters for you.

The reason UMS has ```@Deprecated``` all over the place, is to give the plugin writers time to update their code, but since we have some plugin writers that refuse to do that, most of these will stay like that "forever". But, when you create a new field/variable/method, there's no reason to do this since no plugin can reference something that's new.